src/databricks/labs/lsql/dashboards.py

```diff
@@ -56,6 +56,17 @@
 _SQL_DIALECT = sqlglot.dialects.Databricks
 T = TypeVar("T")
 logger = logging.getLogger(__name__)
+_INVALID_RESOURCE_NAME_MESSAGE = (
+    "Resource names should only contain alphanumeric characters (a-z, A-Z, 0-9), hyphens (-), or underscores (_)"
+)
+
+
+def _is_valid_resource_name(name: str) -> bool:
+    """Verify if the resource name is valid.
+
+    What is valid is defined by the Lakeview API, not by lsql.
+    """
+    return name.replace("-", "").replace("_", "").isalnum()
```

Comment:
Sadly `.isalnum()` uses the Unicode definition of alphanumeric, which means this would pass[^1]: `𝔱𝔥𝔦𝔰-𝔫𝔞𝔪𝔢-𝔦𝔰-𝔫𝔬𝔱-𝔮𝔲𝔦𝔱𝔢-𝔯𝔦𝔤𝔥𝔱`

I think the best way to do this is probably with a regex. (Are there special requirements for the first character? Often there are.)

Something like: `^(?a\w|-)+$` (or exact equivalent `^[A-Za-z0-9_-]+$`)

[^1]: This would make a nice test-case.

Reply:
Thanks, good catch. I've added the test you suggested and updated the check to use regex.